backend/jobsearch/src/main/kotlin/com/jobsearch/exception/GlobalExceptionHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.jobsearch.exception
+
+import com.jobsearch.response.ErrorResponse
+import com.jobsearch.response.StandardResponse
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestControllerAdvice
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
+
+@RestControllerAdvice
+class GlobalExceptionHandler: ResponseEntityExceptionHandler() {
+    @ExceptionHandler(IllegalArgumentException::class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    fun handleIllegalArgumentException(ex: IllegalArgumentException): ResponseEntity<ErrorResponse<Exception>> {
+        return mapResponseEntity(ex, HttpStatus.BAD_REQUEST)
+    }
+
+    @ExceptionHandler(UserAlreadyExistsException::class)
+    fun handleUserAlreadyExistsException(ex: UserAlreadyExistsException): ResponseEntity<ErrorResponse<Exception>>{
+        return mapResponseEntity(ex, HttpStatus.CONFLICT)
+    }
+
+    @ExceptionHandler(NotFoundException::class)
+    fun handleNotFoundException(ex: NotFoundException): ResponseEntity<ErrorResponse<Exception>>{
+        return mapResponseEntity(ex, HttpStatus.NOT_FOUND)
+    }
+
+    @ExceptionHandler(IllegalAccessException::class)
+    fun handleIllegalAccessException(ex: IllegalAccessException): ResponseEntity<ErrorResponse<Exception>>{
+        return mapResponseEntity(ex, HttpStatus.FORBIDDEN)
+    }
+
+    @ExceptionHandler(Exception::class)
+    fun handleAllException(ex: Exception): ResponseEntity<ErrorResponse<Exception>>{
+        return mapResponseEntity(ex, HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
you can remove the second param because the default value for status is `HttpStatus.INTERNAL_SERVER_ERROR`